compiler/rustc_middle/src/ty/consts/int.rs

```diff
@@ -400,6 +400,14 @@ impl ScalarInt {
     }
 }
 
+impl std::ops::Not for ScalarInt {
```

Comment:
this type implements no other operators, so it implementing `Not` seems a bit weird. You can probably just inline this into the caller, or make this an inherent method like `bitwise_not` on the scalarint type.

Reply:
My thinking was it makes sense for unary ops and particularly not (since we don't have to worry about size mismatches and signedness). In the absence of other operators a method probably makes more sense